tests/upsert_validator/test_upsert_validator_vote.py

```diff
@@ -243,15 +243,15 @@ def test_upsert_validator(b, node_key, node_keys, ed25519_node_keys):
                                           new_validator, None).sign([node_key.private_key])
     code, message = b.write_transaction(election, 'broadcast_tx_commit')
     assert code == 202
-    time.sleep(3)
+    time.sleep(1)
```

Comment:
Why do you have a `sleep` here? The commit mode ensures the data is persisted when you get a response.

`sleep`s are inefficient because you always end up waiting for some extra time. If you instead continuously poll the resource in question and fail after a certain timeout, the test only wastes extra time in the unsuccessful case.

Reply:
I tried pulling the `sleep` entirely, and it broke the test. There must be some lag before the tx shows up in the database...

Reply:
Yeah, we need to investigate.

Reply:
It fails because the test calls Tendermint, not BigchainDB (`b.endpoint` points to Tendermint's RPC API). To fix it, fetch validators via `b.get_validators()`.

Reply:
Fixed